frontend-angularjs/app/scripts/controllers/linesCtrl.js

```diff
@@ -1,3 +1,306 @@
+      inboundRoute: inboundRoute,
+      outboundRoute: outboundRoute
+    };
+
+    if (!url) {
+      $http.post(env.backendBaseUrl + "/lines", reqData)
+        .then(function (response) {
+          if (typeof callback === 'function')
+            callback({
+              name: response.data.name,
+              href: response.data._links.line.href
+            });
+        }, function (error) {
+          if (typeof callback === 'function')
+            callback({
+              error: error
+            });
+        });
+    } else {
+      $http.patch(url, reqData)
+        .then(function (response) {
+          if (typeof callback === 'function')
+            callback({
+              name: response.data.name,
+              href: response.data._links.line.href
+            });
+        }, function (error) {
+          if (typeof callback === 'function')
+            callback({
+              error: error
+            });
+        });
+    }
+  };
+
+  $scope.initModal = function () {
+    $.material.init();
+
+    // Init sortable
+    $('.sortable-list').each(function(i, obj) {
+     Sortable.create(obj, {handle: '.list-group-sortable', draggable: ".list-group-sortable-item"});
+    });
+  };
+
+  $scope.fetchStationsContaining = function (name, filterList) {
+    return $http.get(env.backendBaseUrl + "/stations/search/findByNameContainingIgnoreCase?name=" + name)
+      .then(function (response) {
+        if (!filterList)
+          return response.data._embedded.stations;
+
+        return response.data._embedded.stations.filter(function(station) {
+          return filterList.filter(function(filterStation) {
+              return filterStation.name === station.name;
+            }).length === 0;
+        });
+      });
+  };
+
+  $scope.onStationAutocomplete = function ($station, targetArray) {
+    targetArray.push($station);
+  };
+
+  $scope.removeRouteStation = function($index, targetArray) {
+    targetArray.splice($index, 1);
+  };
+
+  $scope.initAgenciesSelect = function () {
+    $http.get(env.backendBaseUrl + "/agencies")
+      .then(function (response) {
+        $scope.agencies = response.data._embedded.agencies;
+      });
+  };
+
+  $scope.closeModal = function () {
+    $uibModalInstance.dismiss('cancel');
+  };
+
+  $scope.mapToUrls = function (stations) {
+    return stations.map(function(station) {
+      return station._links.self.href;
+    });
+  }
+});
+
+
+angular.module('hermesApp').controller('NewLineCtrl', function ($q, $scope, $controller, $http, $uibModalInstance, $timeout, env) {
+  angular.extend(this, $controller('AbstractLineModalCtrl', {$q, $scope: $scope, $uibModalInstance: $uibModalInstance}));
+  var $ctrl = this;
+
+  $scope.line = {};
+  $scope.line.inboundRoute = {stations: []};
+  $scope.line.outboundRoute = {stations: []};
+
+  $scope.saveLine = function () {
+    var inboundRouteUrl, outboundRouteUrl;
+    var inboundRoutePromise = $scope.persistRoute($scope.line.inboundRoute.code,
```

Comment:
Didn't you have 3+ params in your controllers?) In this case I wanted method persistRoute to have all fields in ready to post state. So, I either create RouteDto to pass to method, or pass all params in ready state one by one. By ready state I mean, for example, not stations array as in route object, but station's urls array